### DISCUSSION TITLE
Multiple radio input model binding

### DISCUSSION THREAD

**Discussion by bennoislost** [alpinejs/alpine#2504] [7. Bugs (archived)]
I have component on a page which selects a value from a radio input group. The radio input value is bound to an `x-model` which updates the radio input as checked, which is expected. 

When I add multiple identical components to the same page only the last component instance shows the bound `x-model` when I would expect each instance of the component to reflect the chosen radio input.

Codepen: https://codepen.io/bennoislost/pen/BawWMpL

## Expected behaviour
Radio inputs for every component are bound to the model.

## Actual behaviour
The `x-model` appears to show the correct value but the checked radio input does not reflect that.

## Notes
- Binding `:checked="item.id === selected"` had no effect on the radio input.
- Supplying a unique `x-model` name for each component instance works as expected, but defeats the purpose of reusable components!

**Comment by SimoTod** [comment]
the checked attribute doesn't change the state of a property at runtime (only on page load and Alpine starts after that point). What does it is the checked property (which unfortunately has the same name) so you can theoretically use something like `x-effect="$el.checked = item.id === selected"`.
